tests/test_deprecations.py

```diff
@@ -237,3 +241,12 @@ def test_target_transform_interface():
         numerical.transform(data=pd.DataFrame(columns=["num"]))
     with pytest.warns(DeprecationWarning):
         binary.transform(data=pd.DataFrame(columns=["bin"]))
+
+
+def test_migrated_metadata_attribue():
+    """Accessing the migrated metadata search space attribute raises an error."""
+    with pytest.raises(DeprecationError, match="no longer carry any metadata"):
+        subspace = SubspaceDiscrete.from_parameter(
+            NumericalDiscreteParameter("p", [0, 1])
+        )
+        subspace.metadata()
```

Comment:
technically this is not how users would access the old metadata becasue ti was an attribute

it might be more compliated to make it work proeprly, just wanted raise whether you though about this? can we eg overwrite/expand getattr or so to inject this?

Reply:
no need to do anything complicated here, just the parentheses wrong (potentially some unintended auto-complete or similar), because I implemented the deprecated attribute as a property. In fact, the test was even functional, since the deprecation error would have been raised before attempting to "call" the attribute. But still wrong syntax, of course. Fixed it. Can you have a look again and resolve?

Reply:
gucci